### PR TITLE
[RFR] REST API tests for arbitration_settings

### PR DESCRIPTION
**Pull request by mkoura**
__Adding tests__ for "arbitration_settings" collection.

{{pytest: cfme/tests/test_rest.py -v}}

### CODE REVIEW

**Comment by mkoura** [comment]
@mfalesni

**Comment by dajoRH** [comment]
Lint report for commit 929697662c94f43ba06da0eafb864082a9a83a55:

`cfme/tests/test_rest.py`:
- :bangbang: Line 226:53: **E228** *missing whitespace around modulo operator*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9a85a422067143c8841c3106a559e60f761b0d8e:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/rest/__init__.py [hunk 1]
```diff
@@ -354,3 +355,23 @@ def mark_vm_as_template(rest_api, provider, vm_name):
         lambda: rest_api.collections.templates.find_by(name=vm_name).subcount != 0,
         num_sec=700, delay=15)
     return rest_api.collections.templates.get(name=vm_name)
+
+
+def arbitration_settings(request, rest_api, num=2):
```
Comment (psav) [inline cfme/rest/__init__.py]: I don't like the fact that the rest module is creating the random data. The test should create the random data and pass the ```body``` to the ```arbitration_settings()``` function which then just creates it with whatever the body is.

File: cfme/tests/test_rest.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
 # -*- coding: utf-8 -*-
 """This module contains REST API specific tests."""
 import pytest
+import fauxfactory
+import utils.error as error
 
 from cfme import test_requirements
 from cfme.rest import vm as _vm
+from cfme.rest import arbitration_settings as _arbitration_settings
```
Comment (psav) [inline cfme/tests/test_rest.py]: Why import it with an underscore? Just to keep with the previous one? I don't see a need, but feel free to correct me.

File: cfme/rest/__init__.py [hunk 3]
```diff
@@ -354,3 +355,23 @@ def mark_vm_as_template(rest_api, provider, vm_name):
...
+            'name': 'test_settings_{}'.format(uniq),
+            'display_name': 'Test Settings {}'.format(uniq)})
+    response = collection.action.create(*body)
+
+    @request.addfinalizer
+    def _finished():
+        try:
+            collection.action.delete(*response)
+        except APIException:
+            pass
```
Comment (psav) [inline cfme/rest/__init__.py]: We should at least log that we couldn't clean up

File: cfme/rest/__init__.py (hunk 1 above)
Comment (mkoura) [inline cfme/rest/__init__.py]: @psav This file is collection of helper methods for REST API collections. With few exceptions all functions in this file are creating random data. My understanding of the purpose of these functions is that authors of new tests don't have to figure out how the request for particular collection should look like.

File: cfme/tests/test_rest.py (hunk 2 above)
Comment (mkoura) [inline cfme/tests/test_rest.py]: @psav I thought it's some kind of convention that I don't understand the purpose of :) I'll change that.

File: cfme/rest/__init__.py (hunk 1 above)
Comment (psav) [inline cfme/rest/__init__.py]: Hmmm, OK, then we should rename the module, @mfalesni your input here please

File: cfme/tests/test_rest.py (hunk 2 above)
Comment (psav) [inline cfme/tests/test_rest.py]: Thanks

File: cfme/rest/__init__.py (hunk 1 above)
Comment (mfalesni) [inline cfme/rest/__init__.py]: Possibly. This is how Taras structured it when he was creating the tests. If you look closely, you see it uses request, so this should be used similarly like fixtures, you can import this and then call it from inside a fixture, this was used because sometimes function scope was required and sometimes module scope was enough ...

Comment (mkoura) [inline cfme/rest/__init__.py]: @mfalesni @psav So should I rename it to something like "collections"? Should I do it as separate PR once this one is merged?

Comment (psav) [inline cfme/rest/__init__.py]: ```gen_data``` inside the rest package, ranther than ```__init__```?

**Comment by dajoRH** [comment]
Lint report for commit 27097e5c3a0e7d82fe8ba1dcb4c24ee43b4c1446:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/rest/__init__.py (hunk 1 above)
Comment (mkoura) [inline cfme/rest/__init__.py]: @psav I'll create separate PR for this.
